commit fab6347adfae00d92eb0c6b956bb6f4afaa7ba90
Author:     Sean Yang <seany314@gmail.com>
AuthorDate: Wed Dec 4 13:47:57 2024 -0800
Commit:     GitHub <noreply@github.com>
CommitDate: Wed Dec 4 13:47:57 2024 -0800

    Logger hierarchy (#3081)
    
    * convert to logger hierarchy
    
    * add functions to log_utils
---
 nvflare/private/fed/client/client_engine.py   | 4 ++--
 nvflare/private/fed/client/client_executor.py | 4 ++--
 nvflare/private/fed/server/server_engine.py   | 4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/nvflare/private/fed/client/client_engine.py b/nvflare/private/fed/client/client_engine.py
index c91b9e68..9e7580f0 100644
--- a/nvflare/private/fed/client/client_engine.py
+++ b/nvflare/private/fed/client/client_engine.py
@@ -12,7 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import logging
 import os
 import re
 import shutil
@@ -31,6 +30,7 @@ from nvflare.apis.workspace import Workspace
 from nvflare.fuel.f3.cellnet.cell import Cell
 from nvflare.fuel.f3.cellnet.defs import CellChannel, MessageHeaderKey, ReturnCode
 from nvflare.fuel.f3.message import Message as CellMessage
+from nvflare.fuel.utils.log_utils import get_obj_logger
 from nvflare.private.aux_runner import AuxMsgTarget, AuxRunner
 from nvflare.private.defs import ERROR_MSG_PREFIX, ClientStatusKey, EngineConstant, new_cell_message
 from nvflare.private.event import fire_event
@@ -96,7 +96,7 @@ class ClientEngine(ClientEngineInternalSpec, StreamableEngine):
 
         if workers < 1:
             raise ValueError("workers must >= 1")
-        self.logger = logging.getLogger(self.__class__.__name__)
+        self.logger = get_obj_logger(self)
         self.fl_components = [x for x in self.client.components.values() if isinstance(x, FLComponent)]
 
     def fire_event(self, event_type: str, fl_ctx: FLContext):
diff --git a/nvflare/private/fed/client/client_executor.py b/nvflare/private/fed/client/client_executor.py
index 7cef9903..d7ca4ef4 100644
--- a/nvflare/private/fed/client/client_executor.py
+++ b/nvflare/private/fed/client/client_executor.py
@@ -12,7 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import logging
 import threading
 import time
 from abc import ABC, abstractmethod
@@ -26,6 +25,7 @@ from nvflare.fuel.common.exit_codes import PROCESS_EXIT_REASON, ProcessExitCode
 from nvflare.fuel.f3.cellnet.core_cell import FQCN
 from nvflare.fuel.f3.cellnet.defs import MessageHeaderKey, ReturnCode
 from nvflare.fuel.utils.config_service import ConfigService
+from nvflare.fuel.utils.log_utils import get_obj_logger
 from nvflare.private.defs import CellChannel, CellChannelTopic, JobFailureMsgKey, new_cell_message
 from nvflare.private.fed.utils.fed_utils import get_job_launcher, get_return_code
 from nvflare.security.logging import secure_format_exception, secure_log_traceback
@@ -130,7 +130,7 @@ class JobExecutor(ClientExecutor):
             startup: startup folder
         """
         self.client = client
-        self.logger = logging.getLogger(self.__class__.__name__)
+        self.logger = get_obj_logger(self)
         self.startup = startup
         self.run_processes = {}
         self.lock = threading.Lock()
diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index b13cb62c..f21d8363 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -13,7 +13,6 @@
 # limitations under the License.
 
 import copy
-import logging
 import os
 import re
 import shutil
@@ -52,6 +51,7 @@ from nvflare.fuel.f3.cellnet.defs import ReturnCode as CellMsgReturnCode
 from nvflare.fuel.f3.cellnet.fqcn import FQCN
 from nvflare.fuel.f3.message import Message as CellMessage
 from nvflare.fuel.utils.argument_utils import parse_vars
+from nvflare.fuel.utils.log_utils import get_obj_logger
 from nvflare.fuel.utils.zip_utils import zip_directory_to_bytes
 from nvflare.private.admin_defs import Message, MsgHeader
 from nvflare.private.aux_runner import AuxMsgTarget
@@ -110,7 +110,7 @@ class ServerEngine(ServerEngineInternalSpec, StreamableEngine):
 
         self.executor = ThreadPoolExecutor(max_workers=workers)
         self.lock = Lock()
-        self.logger = logging.getLogger(self.__class__.__name__)
+        self.logger = get_obj_logger(self)
 
         self.asked_to_stop = False
         self.snapshot_persistor = snapshot_persistor
